run.py

```diff
@@ -61,72 +62,90 @@
-        "class": "episode"})
-    podcast_no = len(podcasts) + 1
-    for podcast_li in podcasts:
-        # For each podcast
-        podcast_no -= 1
-
-        # Check podcast not already downloaded
-        download_path = os.path.expanduser(os.path.join(course_dir, f"{podcast_no:02d} - " + podcast_li.a.string +
-                                                        ".mp4"))
-        if os.path.isfile(download_path):
-            print("Skipping podcast", podcast_li.a.string, "(already exists)")
-            continue
+    if get_video_service_podcast_page.status_code != 200:
+        print("Could not get podcast webpage for", name, "- Service responded with status code",
+              get_video_service_podcast_page.status_code)
+        return
 
-        # Podcast not yet downloaded
-        print("Getting podcast", podcast_li.a.string)
+    # Status code valid, parse HTML
+    getVideoServicePodcastPageSoup = BeautifulSoup(get_video_service_podcast_page.content,
+                                                   features="html.parser")
+    podcast_src = "https://video.manchester.ac.uk" + \
+                  getVideoServicePodcastPageSoup.find("video", id="video").source["src"]
 
-        # Get podcast webpage
-        get_video_service_podcast_page = session.get("https://video.manchester.ac.uk" + podcast_li.a["href"])
+    # Get podcast
+    get_video_service_podcast = session.get(podcast_src, stream=True)
 
-        # Check status code valid
-        if get_video_service_podcast_page.status_code != 200:
-            print("Could not get podcast webpage for", podcast_li.a.string, "- Service responded with status code",
-                  get_video_service_podcast_page.status_code)
-            continue
+    # Check status code valid
+    if get_video_service_podcast.status_code != 200:
+        print("Could not get podcast for", name, "- Service responded with status code",
+              get_video_service_podcast.status_code)
+        return
+
+    # Write to file
+    with open(download_path, "wb") as f:
+        get_video_service_podcast.raw.decode_content = True
+        shutil.copyfileobj(get_video_service_podcast.raw, f)
+
+    print("Downloaded podcast", name)
 
-        # Status code valid, parse HTML
-        getVideoServicePodcastPageSoup = BeautifulSoup(get_video_service_podcast_page.content,
-                                                       features="html.parser")
-        podcast_src = "https://video.manchester.ac.uk" + \
-                      getVideoServicePodcastPageSoup.find("video", id="video").source["src"]
 
-        # Get podcast
-        get_video_service_podcast = session.get(podcast_src, stream=True)
+with concurrent.futures.ThreadPoolExecutor(max_workers=settings.concurrent_downloads) as executor:
+    futures = []
+
+    # Status code valid, parse HTML
+    get_video_service_base_soup = BeautifulSoup(get_video_service_base.content, features="html.parser")
+    first = True
```

Comment:
lmao I meant to remove this, I hope you had loads of fun trying to figure out what it was for

Reply:
The height of software development!